### PR TITLE
Manually generate 1.2.1 release from 1.2.0

### PR DESCRIPTION
**Pull request by peternied**
Signed-off-by: Peter Nied <petern@amazon.com>

### Description
Creating a new release manifest for OpenSearch 1.2.1 patch release 

### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
I don't think we should be rebuilding everything from source. If the components don't have changes, use `dist:` with a URL.

**Comment by peternied** [comment]
> I don't think we should be rebuilding everything from source. If the components don't have changes, use `dist:` with a URL.

Just chatted offline, I'll switch to already build dist artifacts for all plugins until we have known replacements.  Lets prove out the patching process by using it here.

File: src/build_workflow/builder_from_dist.py [hunk 1]
```diff
@@ -40,14 +41,28 @@ def export_artifacts(self, build_recorder):
...
                 for artifact in component_manifest.artifacts[artifact_type]:
-                    artifact_url = f"{self.component.dist}/{self.target.platform}/{self.target.architecture}/builds/{self.target_name}/{artifact}"
-                    artifact_dest = os.path.realpath(os.path.join(self.output_path, artifact))
-                    os.makedirs(os.path.dirname(artifact_dest), exist_ok=True)
-                    logging.info(f"Downloading {artifact_url} into {artifact_dest}")
-                    urllib.request.urlretrieve(artifact_url, artifact_dest)
-                    build_recorder.record_artifact(self.component.name, artifact_type, artifact, artifact_dest)
+                    for url_pattern in self.__get_url_patterns():
+                        artifact_url = f"{url_pattern}/{artifact}"
+                        try:
```
Comment (dblock) [inline src/build_workflow/builder_from_dist.py:46]: This swallows too many exceptions and fails silently if a distribution cannot be found. 

1. Add a `distribution_urls` method that returns the list of possible URLs.
2. Add a `distribution_url` method that iterates over `distribution_urls` and returns the first one that has a manifest.yml
3. Use the result from `distribution_url` everywhere.

**Review by dblock — CHANGES_REQUESTED** [review]

File: src/build_workflow/build_args.py [hunk 2]
```diff
@@ -48,6 +48,12 @@ def __init__(self):
             action="store_true",
             help="Do not delete the working temporary directory.",
         )
+        parser.add_argument(
```
Comment (dblock) [inline src/build_workflow/build_args.py:51]: Unused, remove for now?

File: src/manifests/distribution.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import logging
```
Comment (dblock) [inline src/manifests/distribution.py:1]: To be consistent I would make this a class called `Distribution` that takes a base url and product name in the constructor, then add a `find` method.

Needs a test.

File: src/manifests/distribution.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        f"{base_url}/{platform}/{architecture}/builds"
+    ]
+
+    for distribution_url in possible_urls:
+        manifest_url = f"{distribution_url}/manifest.yml"
+        try:
+            with urllib.request.urlopen(manifest_url):
+                # OK we could access the manifest, return the url
+                return distribution_url
+        except:
```
Comment (dblock) [inline src/manifests/distribution.py:24]: Check that the exception is a 403 or 404, fail on everything else.

**Review by dblock — APPROVED** [review]
I'm good with this, we can refactor later. Up to you.

File: src/manifests/distribution.py (hunk 3 above)
Comment (peternied) [reply src/manifests/distribution.py:1]: Lets fast follow on these so the build isn't blocked, https://github.com/opensearch-project/opensearch-build/issues/1289

File: src/manifests/distribution.py (hunk 4 above)
Comment (peternied) [reply src/manifests/distribution.py:24]: Lets fast follow on these so the build isn't blocked, https://github.com/opensearch-project/opensearch-build/issues/1289

**Review by zelinh — APPROVED** [review]
Thanks!

**Pull request merged**